packages/react-magma-dom/src/components/Tabs/TabsScrollSpyContainer.tsx

```diff
@@ -0,0 +1,154 @@
+import React, { useCallback } from 'react';
+import styled from '@emotion/styled';
+import { ScrollSpy } from './utils';
+import { Tabs, TabsOrientation } from './Tabs';
+import { Tab } from './Tab';
+import { TabsContainer } from './TabsContainer';
+import { toCamelCase } from '../../utils';
+
+export interface TabScrollSpyContainerProps
+  extends React.HTMLAttributes<HTMLDivElement> {
+  /*
+   * Adds an icon to the navigation Tab.
+   */
+  icon?: React.ReactElement<any> | React.ReactElement<any>[];
+  isInverse?: boolean;
+  /*
+   * For custom click events.
+   */
+  onClick?: () => void;
+  /**
+   * @internal
+   */
+  testId?: string;
+}
+
+const StyledTabs = styled(Tabs)`
+  flex: 0 auto;
+  position: sticky;
+  top: 0;
+  li {
+    &:after {
+      transition: none;
+    }
+  }
+`;
+
+const StyledContentWrapper = styled.div`
+  flex: 1;
+`;
+
+export const TabsScrollSpyContainer = React.forwardRef<
+  HTMLDivElement,
+  TabScrollSpyContainerProps
+>((props, ref) => {
+  const { children, isInverse, testId, ...other } = props;
+
+  const [options, setOptions] = React.useState([]);
+
+  const [isActive, setIsActive] = React.useState(0);
+
+  const [activeIndex, setActiveIndex] = React.useState(0);
+
+  //Window scroll override
+  React.useEffect(() => {
+    const html = document.querySelector('html');
+    html.style.scrollBehavior = 'auto';
+  }, []);
+
+  //Sets the active tab state on scroll
+  React.useEffect(() => {
+    options.map((option: any) => {
+      // const lastIndex = options.length - 1;
+
+      // const scrollBottom =
+      //   window.innerHeight + window.scrollY >=
+      //   document.documentElement.scrollHeight - 600;
+
+      // if (scrollBottom && activeIndex === lastIndex - 1) {
+      //   setActiveIndex(lastIndex);
+      //   console.log('bottom', lastIndex, typeof option.index);
+      // }
+      if (option.hash === isActive) {
+        setActiveIndex(option.index);
+      }
+    });
+  }, [isActive]);
+
+  const onScrollUpdate = useCallback(
+    (entry, isInViewPort) => {
+      const { target } = entry;
+
+      if (isInViewPort) {
+        setIsActive(toCamelCase(target.id));
+      }
+    },
+    [setIsActive]
+  );
+
+  function onClick(option) {
+    //Prevents smooth page scroll on navigation click
+    window.location.href = `#${option.hash}`;
+    //Allows custom function
+    props.onClick && typeof props.onClick === 'function' && props.onClick();
+  }
+
+  const ScrollSpyNav = ({ options }) => {
+    return (
+      <StyledTabs
+        orientation={TabsOrientation.vertical}
+        testId={testId ? `${testId}-Tabs` : null}
+      >
+        {options.map(option => (
+          <Tab
+            disabled={option.disabled}
+            icon={option.icon}
+            key={option.hash}
+            onClick={() => onClick(option)}
+            data-scrollspy-id={option.hash}
+            testId={testId ? `${testId}-Tab` : null}
```

Comment:
This is using the test id passed to the container, not to the specific tab so they are duplicated:

![image](https://github.com/cengage/react-magma/assets/91160746/3c37417e-8042-4b90-90d5-25ad27ae6cee)

Reply:
I didn't mean it needed to be deleted, I just meant it should be unique. We should be using the testId passed to the panel instead of the container.